### ISSUE TITLE
The character `.` in `Glob.oneOf` causes the string `ERROR` to be returned

### ISSUE THREAD

**Issue by tennety** [dillonkearns/elm-pages#227]
I have a folder structure of
```
content/
  - about.md
  - posts/
      - post1.md
```
My goal is to use `DataSource.Glob` to get the list `[ "about", "posts" ]` so I can build a top-level nav. In `Shared.elm`, I have my `data` defined as:
```elm
type alias Data =
    Set.Set String

data : DataSource.DataSource Data
data =
    (Glob.succeed Basics.identity
        |> Glob.match (Glob.literal "content/")
        |> Glob.capture Glob.wildcard
        |> Glob.match (Glob.oneOf ( ( fileExt, () ), [ ( "/", () ) ] ))
        |> Glob.match Glob.wildcard
        |> Glob.toDataSource
    )
        |> DataSource.map Set.fromList -- Using Set to remove repeats when I have multiple files in a folder
```
If I set `fileExt` as `"."` or `".md"`, everything compiles, but I get the result as `[ "ERROR", "posts" ]`. If I set it to simply `"md"` (no periods in the string), the result is `[ "about.", "posts" ]` (notice the trailing period in the first string). This seems like a parsing issue inside of elm-pages. Anything I'm missing? Happy to provide whatever other context is helpful!

**Comment by dillonkearns** [comment]
Thank you @tennety 🙏  I believe I've got a fix for this in the 3.0 branch. I added an end-to-end test case with your example, so thanks for providing the reproduction!

https://github.com/dillonkearns/elm-pages/blob/c8c7a884da530d6c7694b88193de43f28213d663/examples/end-to-end/src/Test/Glob.elm#L27-L36

**Comment by tennety** [comment]
Sounds great! I can close this 🙏
